### ISSUE TITLE
Integration setup failing

### ISSUE THREAD

**Issue by mdrehn** [PTST/DanskeBiblioteker-HomeAssistant#13]
### Describe the issue

I'm trying to setup the integration for Rudersdal kommune. It accepts my credentials, but goes into error afterwards - giving me the error: "Failed setup, will retry: 400 Bad Request' for url 'https://cover.dandigbib.org/api/v2/covers?type=pid&identifiers=&sizes=small'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400"

### Municipality / library

Rudersdal

### What version of Home Assistant Core has the issue?

2024.10.4

### What type of installation are you running?

Home Assistant OS

### Anything in the logs that might be useful for us?

```text
2024-11-04 20:39:24.757 DEBUG (MainThread) [custom_components.danish_libraries] Updating data
2024-11-04 20:39:24.757 DEBUG (MainThread) [custom_components.danish_libraries] Authenticating
2024-11-04 20:39:27.795 DEBUG (MainThread) [custom_components.danish_libraries] get_loans
2024-11-04 20:39:27.798 DEBUG (MainThread) [custom_components.danish_libraries] get_profile_info
2024-11-04 20:39:27.800 DEBUG (MainThread) [custom_components.danish_libraries] get_reservations
2024-11-04 20:39:27.804 DEBUG (MainThread) [custom_components.danish_libraries] get_ereolen_loans
2024-11-04 20:39:27.806 DEBUG (MainThread) [custom_components.danish_libraries] get_ereolen_reservations
2024-11-04 20:39:28.041 DEBUG (MainThread) [custom_components.danish_libraries.models] {"address": {"coName": "", "street": "Vejvej 1", "postalCode": "1234", "city": "By", "country": "DK"}, "allowBookings": false, "birthday": "1980-01-01", "blockStatus": [], "defaultInterestPeriod": 183, "emailAddress": "email@gmail.com", "name": "First Last", "notificationProtocols": ["BIBAPP"], "onHold": null, "patronId": 24609916, "phoneNumber": null, "preferredLanguage": null, "preferredPickupBranch": "DK-723000", "receiveEmail": true, "receivePostalMail": false, "receiveSms": false, "resident": true, "secondaryAddress": null}
2024-11-04 20:39:28.077 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-04 20:39:28.078 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-04 20:39:28.078 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-04 20:39:28.078 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-04 20:39:28.079 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-04 20:39:28.079 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-04 20:39:28.079 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-04 20:39:28.079 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-04 20:39:28.080 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-04 20:39:28.080 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-04 20:39:28.080 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-04 20:39:28.080 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-04 20:39:28.081 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-04 20:39:28.081 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-04 20:39:28.081 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-04 20:39:28.579 ERROR (MainThread) [custom_components.danish_libraries] Could not extract PID from object, maybe this municipality uses a new url. MUNICIPALITY=<custom_components.danish_libraries.models.library_config.LibraryConfig object at 0x7f581679b0>
2024-11-04 20:39:28.580 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-04 20:39:28.584 ERROR (MainThread) [custom_components.danish_libraries] Could not extract PID from object, maybe this municipality uses a new url. MUNICIPALITY=<custom_components.danish_libraries.models.library_config.LibraryConfig object at 0x7f581679b0>
2024-11-04 20:39:28.584 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-04 20:39:28.604 ERROR (MainThread) [custom_components.danish_libraries] Could not extract PID from object, maybe this municipality uses a new url. MUNICIPALITY=<custom_components.danish_libraries.models.library_config.LibraryConfig object at 0x7f581679b0>
2024-11-04 20:39:28.604 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-04 20:39:28.626 ERROR (MainThread) [custom_components.danish_libraries] Could not extract PID from object, maybe this municipality uses a new url. MUNICIPALITY=<custom_components.danish_libraries.models.library_config.LibraryConfig object at 0x7f581679b0>
2024-11-04 20:39:28.626 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-04 20:39:28.643 ERROR (MainThread) [custom_components.danish_libraries] Could not extract PID from object, maybe this municipality uses a new url. MUNICIPALITY=<custom_components.danish_libraries.models.library_config.LibraryConfig object at 0x7f581679b0>
2024-11-04 20:39:28.643 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-04 20:39:28.650 ERROR (MainThread) [custom_components.danish_libraries] Could not extract PID from object, maybe this municipality uses a new url. MUNICIPALITY=<custom_components.danish_libraries.models.library_config.LibraryConfig object at 0x7f581679b0>
2024-11-04 20:39:28.651 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-04 20:39:28.657 ERROR (MainThread) [custom_components.danish_libraries] Could not extract PID from object, maybe this municipality uses a new url. MUNICIPALITY=<custom_components.danish_libraries.models.library_config.LibraryConfig object at 0x7f581679b0>
2024-11-04 20:39:28.657 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-04 20:39:28.662 ERROR (MainThread) [custom_components.danish_libraries] Could not extract PID from object, maybe this municipality uses a new url. MUNICIPALITY=<custom_components.danish_libraries.models.library_config.LibraryConfig object at 0x7f581679b0>
2024-11-04 20:39:28.662 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-04 20:39:28.694 ERROR (MainThread) [custom_components.danish_libraries] Could not extract PID from object, maybe this municipality uses a new url. MUNICIPALITY=<custom_components.danish_libraries.models.library_config.LibraryConfig object at 0x7f581679b0>
2024-11-04 20:39:28.694 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-04 20:39:28.718 ERROR (MainThread) [custom_components.danish_libraries] Could not extract PID from object, maybe this municipality uses a new url. MUNICIPALITY=<custom_components.danish_libraries.models.library_config.LibraryConfig object at 0x7f581679b0>
2024-11-04 20:39:28.718 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-04 20:39:28.725 ERROR (MainThread) [custom_components.danish_libraries] Could not extract PID from object, maybe this municipality uses a new url. MUNICIPALITY=<custom_components.danish_libraries.models.library_config.LibraryConfig object at 0x7f581679b0>
2024-11-04 20:39:28.725 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-04 20:39:28.735 ERROR (MainThread) [custom_components.danish_libraries] Could not extract PID from object, maybe this municipality uses a new url. MUNICIPALITY=<custom_components.danish_libraries.models.library_config.LibraryConfig object at 0x7f581679b0>
2024-11-04 20:39:28.735 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-04 20:39:28.742 ERROR (MainThread) [custom_components.danish_libraries] Could not extract PID from object, maybe this municipality uses a new url. MUNICIPALITY=<custom_components.danish_libraries.models.library_config.LibraryConfig object at 0x7f581679b0>
2024-11-04 20:39:28.743 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-04 20:39:28.772 ERROR (MainThread) [custom_components.danish_libraries] Could not extract PID from object, maybe this municipality uses a new url. MUNICIPALITY=<custom_components.danish_libraries.models.library_config.LibraryConfig object at 0x7f581679b0>
2024-11-04 20:39:28.772 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-04 20:39:28.779 ERROR (MainThread) [custom_components.danish_libraries] Could not extract PID from object, maybe this municipality uses a new url. MUNICIPALITY=<custom_components.danish_libraries.models.library_config.LibraryConfig object at 0x7f581679b0>
2024-11-04 20:39:28.779 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-04 20:39:29.908 ERROR (MainThread) [custom_components.danish_libraries] Client error '400 Bad Request' for url 'https://cover.dandigbib.org/api/v2/covers?type=pid&identifiers=&sizes=small'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
NoneType: None
2024-11-04 20:39:29.909 ERROR (MainThread) [custom_components.danish_libraries] Client error '400 Bad Request' for url 'https://cover.dandigbib.org/api/v2/covers?type=pid&identifiers=&sizes=small'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
NoneType: None
2024-11-04 20:39:29.909 ERROR (MainThread) [custom_components.danish_libraries] Client error '400 Bad Request' for url 'https://cover.dandigbib.org/api/v2/covers?type=pid&identifiers=&sizes=small'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
NoneType: None
2024-11-04 20:39:29.909 ERROR (MainThread) [custom_components.danish_libraries] Client error '400 Bad Request' for url 'https://cover.dandigbib.org/api/v2/covers?type=pid&identifiers=&sizes=small'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
NoneType: None
2024-11-04 20:39:29.910 ERROR (MainThread) [custom_components.danish_libraries] Client error '400 Bad Request' for url 'https://cover.dandigbib.org/api/v2/covers?type=pid&identifiers=&sizes=small'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
NoneType: None
2024-11-04 20:39:29.910 ERROR (MainThread) [custom_components.danish_libraries] Client error '400 Bad Request' for url 'https://cover.dandigbib.org/api/v2/covers?type=pid&identifiers=&sizes=small'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
NoneType: None
2024-11-04 20:39:29.910 ERROR (MainThread) [custom_components.danish_libraries] Client error '400 Bad Request' for url 'https://cover.dandigbib.org/api/v2/covers?type=pid&identifiers=&sizes=small'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
NoneType: None
2024-11-04 20:39:29.911 ERROR (MainThread) [custom_components.danish_libraries] Client error '400 Bad Request' for url 'https://cover.dandigbib.org/api/v2/covers?type=pid&identifiers=&sizes=small'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
NoneType: None
2024-11-04 20:39:29.911 ERROR (MainThread) [custom_components.danish_libraries] Client error '400 Bad Request' for url 'https://cover.dandigbib.org/api/v2/covers?type=pid&identifiers=&sizes=small'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
NoneType: None
2024-11-04 20:39:29.911 ERROR (MainThread) [custom_components.danish_libraries] Client error '400 Bad Request' for url 'https://cover.dandigbib.org/api/v2/covers?type=pid&identifiers=&sizes=small'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
NoneType: None
2024-11-04 20:39:29.911 ERROR (MainThread) [custom_components.danish_libraries] Client error '400 Bad Request' for url 'https://cover.dandigbib.org/api/v2/covers?type=pid&identifiers=&sizes=small'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
NoneType: None
2024-11-04 20:39:29.912 ERROR (MainThread) [custom_components.danish_libraries] Client error '400 Bad Request' for url 'https://cover.dandigbib.org/api/v2/covers?type=pid&identifiers=&sizes=small'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
NoneType: None
2024-11-04 20:39:29.912 ERROR (MainThread) [custom_components.danish_libraries] Client error '400 Bad Request' for url 'https://cover.dandigbib.org/api/v2/covers?type=pid&identifiers=&sizes=small'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
NoneType: None
2024-11-04 20:39:29.912 ERROR (MainThread) [custom_components.danish_libraries] Client error '400 Bad Request' for url 'https://cover.dandigbib.org/api/v2/covers?type=pid&identifiers=&sizes=small'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
NoneType: None
2024-11-04 20:39:29.912 ERROR (MainThread) [custom_components.danish_libraries] Client error '400 Bad Request' for url 'https://cover.dandigbib.org/api/v2/covers?type=pid&identifiers=&sizes=small'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
NoneType: None
2024-11-04 20:39:29.917 ERROR (MainThread) [custom_components.danish_libraries] Client error '400 Bad Request' for url 'https://cover.dandigbib.org/api/v2/covers?type=pid&identifiers=&sizes=small'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
Traceback (most recent call last):
  File "/config/custom_components/danish_libraries/coordinator.py", line 74, in _async_update_data
    result = coroutine.result()
             ^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 43, in wrapper
    raise e
  File "/config/custom_components/danish_libraries/api.py", line 33, in wrapper
    return await func(*args)
           ^^^^^^^^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 187, in get_loans
    return await self.unpack_results(tasks)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 344, in unpack_results
    return [x.result() for x in done]
            ^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 43, in wrapper
    raise e
  File "/config/custom_components/danish_libraries/api.py", line 33, in wrapper
    return await func(*args)
           ^^^^^^^^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 291, in get_info
    image_url = await self.get_image_cover(pid, "pid")
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 43, in wrapper
    raise e
  File "/config/custom_components/danish_libraries/api.py", line 33, in wrapper
    return await func(*args)
           ^^^^^^^^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 333, in get_image_cover
    image_response.raise_for_status()
  File "/usr/local/lib/python3.12/site-packages/httpx/_models.py", line 763, in raise_for_status
    raise HTTPStatusError(message, request=request, response=self)
httpx.HTTPStatusError: Client error '400 Bad Request' for url 'https://cover.dandigbib.org/api/v2/covers?type=pid&identifiers=&sizes=small'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
2024-11-04 20:39:29.921 ERROR (MainThread) [custom_components.danish_libraries] Unexpected error fetching danish_libraries data
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 354, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/danish_libraries/coordinator.py", line 74, in _async_update_data
    result = coroutine.result()
             ^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 43, in wrapper
    raise e
  File "/config/custom_components/danish_libraries/api.py", line 33, in wrapper
    return await func(*args)
           ^^^^^^^^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 187, in get_loans
    return await self.unpack_results(tasks)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 344, in unpack_results
    return [x.result() for x in done]
            ^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 43, in wrapper
    raise e
  File "/config/custom_components/danish_libraries/api.py", line 33, in wrapper
    return await func(*args)
           ^^^^^^^^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 291, in get_info
    image_url = await self.get_image_cover(pid, "pid")
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 43, in wrapper
    raise e
  File "/config/custom_components/danish_libraries/api.py", line 33, in wrapper
    return await func(*args)
           ^^^^^^^^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 333, in get_image_cover
    image_response.raise_for_status()
  File "/usr/local/lib/python3.12/site-packages/httpx/_models.py", line 763, in raise_for_status
    raise HTTPStatusError(message, request=request, response=self)
httpx.HTTPStatusError: Client error '400 Bad Request' for url 'https://cover.dandigbib.org/api/v2/covers?type=pid&identifiers=&sizes=small'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
2024-11-04 20:39:29.924 DEBUG (MainThread) [custom_components.danish_libraries] Finished fetching danish_libraries data in 5.168 seconds (success: False)
```

**Comment by PTST** [comment]
This should be fixed in the latest v0.1.6 version 🙂

**Comment by mdrehn** [comment]
Thanks for quick fix! Now it seems to get stuck when the integration is initializing. I have tried to disable and re-enable it, but it keeps saying "initializing". 🙁

**Comment by PTST** [comment]
Can you share your logs again?  
Do you know if you have any reservations or loans right now?  
If possible could you share a screenshot of your current loans and reservations from the library website? 😊

**Comment by mdrehn** [comment]
Yes, sure - this is log from when I re-enable the integration and a couple of minutes in. As mentioned, it keep saying "Initializing" and never proceeds from there.

I do not have any reservations, but I do have current loans. See below for screenshot.

2024-11-05 17:48:13.860 DEBUG (MainThread) [custom_components.danish_libraries] Updating data
2024-11-05 17:48:13.861 DEBUG (MainThread) [custom_components.danish_libraries] Authenticating
2024-11-05 17:48:17.040 DEBUG (MainThread) [custom_components.danish_libraries] get_loans
2024-11-05 17:48:17.043 DEBUG (MainThread) [custom_components.danish_libraries] get_profile_info
2024-11-05 17:48:17.045 DEBUG (MainThread) [custom_components.danish_libraries] get_reservations
2024-11-05 17:48:17.048 DEBUG (MainThread) [custom_components.danish_libraries] get_ereolen_loans
2024-11-05 17:48:17.050 DEBUG (MainThread) [custom_components.danish_libraries] get_ereolen_reservations
2024-11-05 17:48:17.328 DEBUG (MainThread) [custom_components.danish_libraries.models] {"address": {"coName": "", "street": "Vejvej 1", "postalCode": "1234", "city": "By", "country": "DK"}, "allowBookings": false, "birthday": "1980-01-01", "blockStatus": [], "defaultInterestPeriod": 183, "emailAddress": "email@gmail.com", "name": "First Last", "notificationProtocols": ["BIBAPP"], "onHold": null, "patronId": 24609916, "phoneNumber": null, "preferredLanguage": null, "preferredPickupBranch": "DK-723000", "receiveEmail": true, "receivePostalMail": false, "receiveSms": false, "resident": true, "secondaryAddress": null}
2024-11-05 17:48:17.423 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-05 17:48:17.423 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-05 17:48:17.424 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-05 17:48:17.424 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-05 17:48:17.424 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-05 17:48:17.424 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-05 17:48:17.425 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-05 17:48:17.425 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-05 17:48:17.425 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-05 17:48:17.425 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-05 17:48:17.425 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-05 17:48:17.426 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-05 17:48:17.426 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-05 17:48:17.426 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-05 17:48:17.427 DEBUG (MainThread) [custom_components.danish_libraries] get_info
2024-11-05 17:48:18.213 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-05 17:48:18.222 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-05 17:48:18.461 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-05 17:48:18.585 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-05 17:48:18.601 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-05 17:48:19.986 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-05 17:48:20.015 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-05 17:48:20.025 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-05 17:48:20.057 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-05 17:48:20.073 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-05 17:48:20.097 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-05 17:48:20.122 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-05 17:48:20.146 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-05 17:48:20.155 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-05 17:48:20.177 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-05 17:48:20.632 DEBUG (MainThread) [custom_components.danish_libraries] list index out of range
2024-11-05 17:48:20.632 DEBUG (MainThread) [custom_components.danish_libraries] Unknown error, retrying in 30sec
Traceback (most recent call last):
  File "/config/custom_components/danish_libraries/api.py", line 33, in wrapper
    return await func(*args)
           ^^^^^^^^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 331, in get_image_cover
    image_json = image_response.json()[0]
                 ~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
2024-11-05 17:48:50.949 DEBUG (MainThread) [custom_components.danish_libraries] list index out of range
2024-11-05 17:48:50.949 DEBUG (MainThread) [custom_components.danish_libraries] Unknown error, retrying in 30sec
Traceback (most recent call last):
  File "/config/custom_components/danish_libraries/api.py", line 33, in wrapper
    return await func(*args)
           ^^^^^^^^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 331, in get_image_cover
    image_json = image_response.json()[0]
                 ~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/config/custom_components/danish_libraries/api.py", line 33, in wrapper
    return await func(*args)
           ^^^^^^^^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 288, in get_info
    image_url = await self.get_image_cover(pid, "pid")
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 53, in wrapper
    return await func(*args)
           ^^^^^^^^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 331, in get_image_cover
    image_json = image_response.json()[0]
                 ~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
2024-11-05 17:49:21.399 DEBUG (MainThread) [custom_components.danish_libraries] get_image_cover
2024-11-05 17:49:21.608 DEBUG (MainThread) [custom_components.danish_libraries] list index out of range
2024-11-05 17:49:21.609 DEBUG (MainThread) [custom_components.danish_libraries] Unknown error, retrying in 30sec
Traceback (most recent call last):
  File "/config/custom_components/danish_libraries/api.py", line 33, in wrapper
    return await func(*args)
           ^^^^^^^^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 331, in get_image_cover
    image_json = image_response.json()[0]
                 ~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/config/custom_components/danish_libraries/api.py", line 33, in wrapper
    return await func(*args)
           ^^^^^^^^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 288, in get_info
    image_url = await self.get_image_cover(pid, "pid")
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 53, in wrapper
    return await func(*args)
           ^^^^^^^^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 331, in get_image_cover
    image_json = image_response.json()[0]
                 ~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/config/custom_components/danish_libraries/api.py", line 33, in wrapper
    return await func(*args)
           ^^^^^^^^^^^^^^^^^
  File "/config/custom_components/danish_libraries/api.py", line 331, in get_image_cover
    image_json = image_response.json()[0]
                 ~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range


![image](https://github.com/user-attachments/assets/df289f88-c23d-4265-a8b9-68e5f97dbe8f)

**Comment by PTST** [comment]
I can see that it's now failing because it can't find the cover image for at least one of the books.  
I have released 0.1.7 which handles this more gracefully, by having a default image for when this occurs.  
After you have installed it could you please give me a list of which books show up with this as their cover? I would like to look them up manually on the rudersdal website to see why these behave differently  
![cover-image](https://upload.wikimedia.org/wikipedia/commons/thumb/6/65/No-Image-Placeholder.svg/128px-No-Image-Placeholder.svg.png)

**Comment by PTST** [comment]
Great 👍 I'll close the issue then
